sklearn/manifold/t_sne.py

```diff
@@ -433,24 +639,51 @@ def _fit(self, X):
         else:
             if self.verbose:
                 print("[t-SNE] Computing pairwise distances...")
-            
             if self.metric == "euclidean":
-                distances = pairwise_distances(X, metric=self.metric, squared=True)
+                distances = pairwise_distances(X, metric=self.metric,
+                                               squared=True)
             else:
                 distances = pairwise_distances(X, metric=self.metric)
 
+        assert np.all(distances >= 0), "All distances should be positive"
```

Comment:
you should probably raise a ValueError in this case, either the metric or the distances given as X are not correct